### DISCUSSION TITLE
[BUG] Listview Does Not Display the Correct Number of Lines Set in Config.rasi

### DISCUSSION THREAD

**Discussion by denppa** [davatorium/rofi#1788] [Theming] [answered]
### Rofi version (rofi -v)

Version: 1.7.5

### Configuration

https://gist.github.com/Horuyuki/89fd021369aa92c954fa0d3a6e24970b

### Theme

https://gist.github.com/Horuyuki/5a0cf72ec81871d26f87d123d0db012a

### Timing report

G_MESSAGES_DEBUG=Timings set. bash: set.: command not found

### Launch command

rofi -show drun

### Step to reproduce

1. Use the above theme config
2. Launch rofi with the above command
3. View incorrect render

### Expected behavior

1. show the number of lines I set in listview
2. There should not be any *chin* / empty space under list view when it is not specified.

### Actual behavior

1. The listview has 3 lines instead of 2
2. There is a chin under the list.

### Additional information

https://i.imgur.com/S4UnW0j.png

Arch linux
`6.1.6-arch1-1`
picom
i3wm

Please let me know if you need any information. I did see this issue in another thread, but I did not install rofi from build files, but rather the repos, therefore I do not know how to patch it.

Issue that mentioned it: https://github.com/davatorium/rofi/issues/1190

### Using wayland display server protocol

- [X] No, I don't use the wayland display server protocol

### Update 1
Setting the width and height to 40% helped the chin problem, but there is still a little bit left, and the number shown is still incorrect.

https://i.imgur.com/VG30v1h.png

**Comment by DaveDavenport** [comment, marked as answer]
You have conflicting constraints.  You set a height and set number of lines.  If you set height, the number of lines is based on how many rows fit in the space given to the listview.
(making height of all widgets 30% is a weird requirement)

**Comment by DaveDavenport** [comment]
![rofi-2023-01-22-0512-00000](https://user-images.githubusercontent.com/406245/213900259-6c2b39f3-d910-43a1-b116-933073e3daa8.png)
removing height fixes it for me.. 2 lines as requested.

You might also want to set `vertical-align: 0.5;` on the element-text to center the text vertically..

**Reply by denppa** [reply]
Thank you very much, it is indeed displaying correctly right now!

I copied the config from someone else, so the adjustments I made might not necessarily make sense.

Vertical align on the text is a beautiful suggestion too.

**Reply by DaveDavenport** [reply]
Great, good to hear. please mark the discussion as answered if it is resolved :smile:
